awx/main/models/events.py

```diff
@@ -510,12 +510,15 @@ def _update_host_summary_from_stats(self, hostnames):
             job = self.job
 
             from awx.main.models import Host, JobHostSummary  # circular import
+            from awx.main.models import Host, JobHostSummary, HostMetric
 
             all_hosts = Host.objects.filter(pk__in=self.host_map.values()).only('id')
             existing_host_ids = set(h.id for h in all_hosts)
 
             summaries = dict()
+            updated_hosts_list = list()
             for host in hostnames:
+                updated_hosts_list.append(host)
```

Comment:
how is `updated_hosts_list` any different from `hostnames`?

Reply:
I'm increasingly thinking that this shouldn't be needed.

Reply:
I think because `hostnames` comes in as a `set`

Reply:
The Django ORM is just as happy with a `set` as a `list` (and the ordering is equally arbitrary in both cases). The introduction of `updated_hosts_list` should be removed, and the later use of it should simply be replaced with `hostnames`.